fiboa_cli/convert_utils.py

```diff
@@ -466,3 +132,402 @@ def read_geojson(path, **kwargs):
     obj["features"] = list(map(normalize_geojson_properties, obj["features"]))
 
     return gpd.GeoDataFrame.from_features(obj, crs = "EPSG:4326")
+
+
+class BaseConverter:
+    bbox: Optional[tuple[float]] = None
+    id: str = None
+    short_name: str = None
+    title: str = None
+    license: str | dict[str, str] = None
+    attribution: str = None
+    description: str = None
+    providers: list[dict] = None
+
+    sources: Optional[dict[str, str] | str] = None
+    source_variants: Optional[dict[dict[str, str] | str]] = None
+    variant: str = None
+
+    columns: dict[str, str] = None
+    column_additions: dict[str, str] = EMPTY_DICT
+    column_filters: dict[str, callable] = EMPTY_DICT
+    column_migrations: dict[str, callable] = EMPTY_DICT
+    missing_schemas: dict[str, str] = EMPTY_DICT
+    extensions: list[str] = tuple()
+
+    index_as_id = False
+
+    def __init__(self, **kwargs):
+        # This init method allows you to override all properties & methods
+        # It's a bit hacky but allows a gradual conversion from function-based to class-based converters
+        self.__dict__.update({k: v for k, v in kwargs.items() if v is not None})
+        for key in ("id", "short_name", "title", "license", "columns"):
+            assert getattr(self, key) is not None, f"{self.__class__.__name__} misses required attribute {key}"
+
+    @property
+    def ID(self):  # noqa backwards compatibility for function-based converters
+        return self.id
+
+    def migrate(self, gdf):
```

Comment:
If I'm not mistaken:

```suggestion
    def migrate(self, gdf) -> gpd.GeoDataFrame:
```

Reply:
Will do